lib/internal/url.js

```diff
@@ -625,7 +729,7 @@ class URL {
     this.#context.port = port;
     this.#context.hash = hash;
     if (this.#searchParams) {
-      this.#searchParams[searchParams] = parseParams(search);
+      setURLSearchParamsList(this.#searchParams, parseParams(search));
```

Comment:
Can we call this function, parseParams, inside this static function? I couldn't find a usage where we dont need a parseParams function unless it is [] in updateContext. It would make it more maintainable(imho)

Reply:
And we can make parseParams a private function?

Reply:
I'm not totally sure what you are suggesting. Can you show the end result of your suggestion?

Reply:
parseParams doesn't need to be a function on it's own. We can move it inside `URLSearchParams.prototype.#parseParams` and update the other implementations like:

For setting it:

`setURLSearchParamsList(this.#searchParams, search)`

To remove it (for example inside `updateContext()` function):

`setURLSearchParamsList(this.#searchParams, undefined)`

Reply:
Ah okay. I'd prefer leaving `parseParams` as a standalone function, since in the spec it's not tied to URLSearchParams specifically. However I can definitely make `setURLSearchParamsList` take a string, rather than requiring the user to parse the parameters themselves.